vcd/resource_vcd_access_control_vapp.go

```diff
@@ -0,0 +1,355 @@
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vappId := d.Get("vapp_id").(string)
+	vapp, err := vdc.GetVAppByNameOrId(vappId, false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp. %s", err)
+	}
+	err = vapp.RemoveAccessControl()
+	if err != nil {
+		return fmt.Errorf("error removing access control for vApp %s: %s", vapp.VApp.Name, err)
+	}
+	d.SetId("")
+	return nil
+}
+
+func accessControlVappImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 3 {
+		return nil, fmt.Errorf("[vApp access control import] resource identifier must be specified as org.vdc.my-vapp")
+	}
+	listRequested := false
+	orgName, vdcName, vappIdentifier := resourceURI[0], resourceURI[1], resourceURI[2]
+	if strings.HasPrefix(orgName, "list@") {
+		listRequested = true
+		orgNameList := strings.Split(orgName, "@")
+		if len(orgNameList) < 2 {
+			return nil, fmt.Errorf("[vApp access control import] empty Org name provided with list@ request")
+		}
+		orgName = orgNameList[1]
+	}
+	if orgName == "" {
+		return nil, fmt.Errorf("[vApp access control import] empty org name provided")
+	}
+	if vdcName == "" {
+		return nil, fmt.Errorf("[vApp access control import] empty VDC name provided")
+	}
+	if vappIdentifier == "" {
+		return nil, fmt.Errorf("[vApp access control import] empty vApp access control identifier provided")
+	}
+
+	vcdClient := meta.(*VCDClient)
+	adminOrg, err := vcdClient.GetAdminOrgByName(orgName)
+	if err != nil {
+		return nil, fmt.Errorf(errorRetrievingOrg, orgName)
+	}
+
+	vdc, err := adminOrg.GetVDCByName(vdcName, false)
+	if err != nil {
+		return nil, fmt.Errorf("[vApp access control import] error retrieving VDC %s, %s", vdcName, err)
+	}
+
+	if listRequested {
+		var vappList []*types.ResourceReference = vdc.GetVappList()
+		return nil, fmt.Errorf("[vApp access control import] list of all vApps:\n%s", formatVappList(vappList))
+	}
+
+	var vapp *govcd.VApp
+	vapp, err = vdc.GetVAppByNameOrId(vappIdentifier, false)
+
+	if err != nil {
+		return nil, fmt.Errorf("[vApp access control import] error retrieving vapp %s: %s",
+			vappIdentifier, err)
+	}
+
+	_ = d.Set("org", orgName)
+	_ = d.Set("vdc", vdcName)
+	_ = d.Set("vapp_id", vapp.VApp.ID)
+	d.SetId(vapp.VApp.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func formatVappList(list []*types.ResourceReference) string {
+	result := ""
+	for i, ref := range list {
+		result += fmt.Sprintf("%3d %-30s %s\n", i, ref.Name, ref.ID)
+	}
+	return result
+}
+
+func accessControlListToSharedSet(input []*types.AccessSetting) ([]map[string]interface{}, error) {
+	var output []map[string]interface{}
+
+	for _, item := range input {
+		var setting = make(map[string]interface{})
+
+		switch item.Subject.Type {
+		case types.MimeAdminUser:
+			setting["user_id"] = "urn:vcloud:user:" + extractUuid(item.Subject.HREF)
```

Comment:
Why here we have full ID and other only UUID

Reply:
The API returns the full ID. If I use the UUID only here, we'll have a discrepancy in Terraform state.